src/security/passwords.py

```
@@ -1,10 +1,59 @@
+import os
+from datetime import datetime, timedelta
+
+from jose import JWTError, jwt
 from passlib.context import CryptContext
+from sqlalchemy import select
+
+from security.token_manager import JWTAuthManager
+
+pwd_context = CryptContext(schemes=["bcrypt"], bcrypt__rounds=14, deprecated="auto")
+
+
+SECRET_KEY = "mysecretkey"
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+
+def create_access_token(data: dict, expires_delta: timedelta = None):
+    to_encode = data.copy()
+    if expires_delta:
+        expire = datetime.utcnow() + expires_delta
+    else:
+        expire = datetime.utcnow() + timedelta(minutes=15)
+    to_encode.update({"exp": expire})
+    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
+    return encoded_jwt
+
+
+def decode_token(token: str):
+    try:
+        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
+        return payload
+    except JWTError:
+        return None
```

`decode_token` swallows JWT errors and returns `None`. In many call sites it's better to raise a clear exception or re-raise a custom error so callers can differentiate between an invalid token and other failures. If you keep returning `None`, ensure all callers correctly check for `None` and react with proper HTTP errors. Document this behavior explicitly.